lxd/daemon.go

```diff
@@ -669,6 +674,88 @@ func (d *Daemon) handleOIDCAuthenticationResult(r *http.Request, result *oidc.Au
 	return nil
 }
 
+// getSecrets gets a copy of the current, cluster-wide secrets.
+func (d *Daemon) getSecrets(ctx context.Context) (dbCluster.AuthSecrets, error) {
```

Comment:
in general this function is too complex and should be broken up, starting with at least moving the DB queries to functions with the other DB query functions.

Reply:
Made this much simpler. The `(AuthSecrets).Rotate` function now accepts a transaction and will perform the rotation in-memory and in the database at once.